cmake/sdlchecks.cmake

```diff
@@ -662,6 +662,22 @@ macro(CheckWayland)
         set(EXTRA_LIBS ${WAYLAND_LIBRARIES} ${EXTRA_LIBS})
       endif()
 
+      if(WAYLAND_LIBDECORATION)
+        include(ExternalProject)
+        ExternalProject_Add(libdecoration
+          PREFIX "${CMAKE_CURRENT_BINARY_DIR}/extern/libdecoration"
+          GIT_REPOSITORY https://gitlab.gnome.org/jadahl/libdecoration.git
+          GIT_TAG master
+          CONFIGURE_COMMAND meson --prefix "${CMAKE_CURRENT_BINARY_DIR}/install" --libdir "lib" ../libdecoration
+          BUILD_COMMAND ninja
+          INSTALL_COMMAND ninja install
+        )
+
+        set(SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_LIBDECORATION "\"libdecoration.so\"")
```

Comment:
I would _really_ like to not have this pull in external code at configure time, especially if we have to count on meson being installed to make this work.

Generally we check if the header/lib is available on the system and either enable the feature, or not, from there.

Is libdecoration not available in distros yet?

Reply:
At the moment libdecoration is still in development, the ABI hasn't frozen yet. This probably won't be in distributions until 1.0 is tagged.

Reply:
I agree that the final merged support should not pull in the dependency like this. How is SDL handling third-party dependencies that are not distributed by distributions yet? In the age of flatpak and snaps, this is not an issue since these third-party dependencies can be pulled in manually from source.

`libdecoration` has not had a release yet. Some pending PRs are to be merged and a 0.1 release is imminent (cc @jadahl). But even then, it will take a while until this is picked up by distributions.

@icculus What do you propose?
- Do you want to wait until a first libdecoration 0.1 release is distributed by distributions, before supporting Wayland? If so, which distributions would that be?
- Should there be a way for SDL to depend on third-party libraries?

Reply:
The SDL source tree already contains it's own set of `wayland-protocols` instead of depending on the definitions supplied by the distributions. Should all third-party dependencies be integrated this way directly in the source three?

Reply:
To be clear, I'm ignorant of obvious details here, so:

- If libdecoration is just a C callable API, we will require the headers be available on the system at build time and will dynamically load the shared library at runtime. If it's not in distributions yet, that's not our problem; if we see it when configuring, it's enabled, otherwise it's not. This lets the bleeding edge install it from source before the world catches up, but the binary that is built will work with or without the library on the end-user's system.
- If it's just a wayland protocol, that's a no-brainer, we just put the protocol files in SDL's tree and call it a day. At runtime we either have the protocol or we don't and we have to handle both cases anyhow. Assuming they're doing things in the usual Wayland way, we can commit before ABI is stable because the protocol name will change and we'll ignore the changed version (and later, we can decide to upgrade or support both codepaths as appropriate...for this it sounds like we'd just dump support for the pre-0.1 release).

But it would be helpful to for @jadahl to say how far from a stable ABI we are at the moment.

Reply:
Libdecoration is C API/ABI, that aims to lock into an ABI in the near future. I intend to package it for Fedora (F35) after that happens, and I hope it'll show up elsewhere as well. I'm sure you will see comments about it here when that happens.